grub/02_coreos_firstboot

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+exec tail -n +3 $0
+set coreos_firstboot=""
+# Determine if this is a first boot.
+if [ -f "/coreos-firstboot" ]; then
```

Comment:
Hi, I am still a bit confused about `/coreos-firstboot` vs `/boot/coreos-firstboot` after reading the comment:
```
the mistake we made in the grub fragment is having /boot/coreos-firstboot instead of just /coreos-firstboot, because we have a separate partition in the ostree-based systems we ship
```

The confusing part for me is that this place seems to be the only place that we check for `/coreos-firstboot` directory , and other places all seem to use `/boot/coreos-firstboot`. Hmm, I seemed to be missing something, will `/$dir` be referenced/changed to `/boot/$dir` somehow on ostree-based systems? ( asking out of curiosity :p)

Reply:
It's because in grub, `/` is the boot partition - although that's *if* a boot partition exists.  Rahul was originally testing on a classic Fedora Cloud system, where there isn't a separate partition.  Long term...we want to support not having `/boot` be a separate partition in some scenarios probably.  There's an old [ostree issue](https://github.com/ostreedev/ostree/issues/1452) for this, but doing so gets more complicated with Ignition-disks potentially in the loop.